scrapy/http/request/form.py

```diff
@@ -35,6 +44,8 @@ def __init__(self, *args, **kwargs):
             else:
                 self._set_url(self.url + ('&' if '?' in self.url else '?') + querystr)
 
+    formdata = property(_get_formdata, obsolete_setter(_set_formdata, 'formdata'))
```

Comment:
why should we use obsolete_setter in new code?

Reply:
I think there is a problem with API: obvious `request.formdata['foo'] = 'bar'` won't work as expected. And if we want to make it work, a naive solution can be bad for the performance (body can be rebuilt after each change to formdata).